apps/emqx_bridge_azure_blob_storage/src/emqx_bridge_azure_blob_storage_action_schema.erl

```diff
@@ -0,0 +1,310 @@
+
+%% `hocon_schema' API
+-export([
+    namespace/0,
+    roots/0,
+    fields/1,
+    desc/1
+]).
+
+%% `emqx_bridge_v2_schema' "unofficial" API
+-export([
+    bridge_v2_examples/1
+]).
+
+%% API
+-export([]).
+
+%%------------------------------------------------------------------------------
+%% Type declarations
+%%------------------------------------------------------------------------------
+
+%%-------------------------------------------------------------------------------------------------
+%% `hocon_schema' API
+%%-------------------------------------------------------------------------------------------------
+
+namespace() ->
+    "action_azure_blob_storage".
+
+roots() ->
+    [].
+
+fields(Field) when
+    Field == "get_bridge_v2";
+    Field == "put_bridge_v2";
+    Field == "post_bridge_v2"
+->
+    emqx_bridge_v2_schema:api_fields(Field, ?ACTION_TYPE, fields(?ACTION_TYPE));
+fields(action) ->
+    {?ACTION_TYPE,
+        mk(
+            hoconsc:map(name, hoconsc:ref(?MODULE, ?ACTION_TYPE)),
+            #{
+                desc => <<"Azure Blob Storage Action Config">>,
+                required => false
+            }
+        )};
+fields(?ACTION_TYPE) ->
+    emqx_bridge_v2_schema:make_producer_action_schema(
+        mk(
+            mkunion(mode, #{
+                <<"direct">> => ref(direct_parameters),
+                <<"aggregated">> => ref(aggreg_parameters)
+            }),
+            #{
+                required => true,
+                desc => ?DESC("parameters")
+            }
+        ),
+        #{
+            resource_opts_ref => ref(action_resource_opts)
+        }
+    );
+fields(direct_parameters) ->
+    [
+        {mode, mk(direct, #{required => true, desc => ?DESC("direct_mode")})},
+        %% Container in the Azure Blob Storage domain, not aggregation.
+        {container,
+            mk(
+                emqx_schema:template_str(),
+                #{
+                    desc => ?DESC("direct_container_template"),
+                    required => true
+                }
+            )},
+        {blob,
+            mk(
+                emqx_schema:template_str(),
+                #{
+                    desc => ?DESC("direct_blob_template"),
+                    required => true
+                }
+            )},
+        {content,
+            mk(
+                emqx_schema:template(),
+                #{
+                    required => false,
+                    default => <<"${.}">>,
+                    desc => ?DESC("direct_content_template")
+                }
+            )}
+        | fields(common_action_parameters)
+    ];
+fields(aggreg_parameters) ->
+    [
+        {mode, mk(aggregated, #{required => true, desc => ?DESC("aggregated_mode")})},
+        {aggregation, mk(ref(aggregation), #{required => true, desc => ?DESC("aggregation")})},
+        %% Container in the Azure Blob Storage domain, not aggregation.
+        {container,
+            mk(
+                string(),
```

Comment:
?
```suggestion
                binary(),
```

Reply:
This must be a `string()` for `erlazure`.